engine/src/main/java/org/destinationsol/game/ship/SolShip.java

```diff
@@ -417,7 +418,8 @@ private void throwLoot(SolGame game, SolItem item, boolean onDeath) {
 
     @Override
     public void receiveDmg(float dmg, SolGame game, Vector2 position, DmgType dmgType) {
-        if (dmg <= 0) {
+        Hero hero = game.getHero();
+        if (dmg <= 0 || (hero.isInvincible() && hero.getShip() == this)) {
```

Comment:
I assume that you know what you're doing with `==` as opposed to `.equals()` :+1: